File: TOI/TOI15/toi15_medal.cpp
#include<bits/stdc++.h> 
using namespace std ;
int a[500001];
int b[500001];
bool cmp(int a, int b){
    return a > b ;
}
int main(){
    int n ; cin >> n ;
    int MAX  = INT_MIN ;
    int MIN  = INT_MAX ;
    for(int i = 0 ; i < n ; i ++ ){
        cin >> a[i] ;
    }
    for(int i = 0 ; i < n ; i ++ ){
        cin >> b[i] ;
    }
    sort(a , a + n);
    sort(b , b + n , cmp);
    for(int i = 0 ; i < n ; i ++ ){
        int now = a[i] + b[i] ;
        MAX = max(MAX , now) ;
        MIN = min(MIN , now) ;
    }
    cout << MAX - MIN ;
    return 0 ;
}

File: TOI/TOI15/toi15_medal.py
n = int(input())
a = input().split(' ')
b = input().split(' ')
a = [int(i) for i in a]
b = [int(i) for i in b]

a.sort()
b.sort(reverse=True)

MAX = 0 
MIN = 10000000000 

for i in range(0 , n) :
    sum = a[i] + b[i] 
    MAX = max(MAX , sum) 
    MIN = min(MIN , sum) 

print(MAX - MIN) 
    
